### PR TITLE
Type warnings

### PR DESCRIPTION
**Pull request by BridgeAR**
This is on top of #3836 and I wanted to get your feedback what you think about the implementation. So you only have to look at the "Initial commit" changes.

If someone uses a data type in a way that is not supported we'd now warn the user about it including what we do instead and presenting the link to the data types as they are in the current dialect.

### CODE REVIEW

**Comment by janmeier** [comment]
I like this implementation very much! :) LGTM

**Comment by BridgeAR** [comment]
This will close #2038

File: lib/dialects/mssql/data-types.js [hunk 1]
```diff
@@ -68,13 +76,85 @@ DATE.prototype.toSql = function() {
   return 'DATETIME2';
 };
 
+var INTEGER = function() {
+  if (!(this instanceof INTEGER)) return new INTEGER();
+  BaseTypes.INTEGER.apply(this, arguments);
+
+  // MSSQL does not support any parameters for integer
+  if (this._length || this.options.length || this._unsigned || this._zerofill) {
+    this.warn('You used INTEGER with a attribute. This is not supported by MSSQL. Simple `INTEGER` will be used instead.');
```
Comment (mickhansen) [inline lib/dialects/mssql/data-types.js]: 'an attribute'

File: lib/data-types.js [hunk 2]
```diff
@@ -38,12 +39,20 @@ var ABSTRACT = function(options) {
...
 ABSTRACT.prototype.toString = function() {
   return this.toSql();
 };
 ABSTRACT.prototype.toSql = function() {
   return this.key;
 };
+ABSTRACT.prototype.warn = function(text) {
+  if (!warnings[text]) {
+    warnings[text] = true;
+    console.warn('>> WARNING:', text, '\n>> Check:', this.dialectTypes);
```
Comment (mickhansen) [inline lib/data-types.js]: Hmm, any way to have this include the actual line ala what depd does you think?

File: lib/dialects/postgres/data-types.js [hunk 3]
```diff
@@ -55,10 +65,13 @@ var INTEGER = function() {
   BaseTypes.INTEGER.apply(this, arguments);
 
   // POSTGRES does not support any parameters for integer
-  this._length = null;
-  this.options.length = null;
-  this._unsigned = null;
-  this._zerofill = null;
+  if (this._length || this.options.length || this._unsigned || this._zerofill) {
+    this.warn('You used INTEGER with a parameter. This is not supported by POSTGRES. Simple `INTEGER` will be used instead.');
```
Comment (mickhansen) [inline lib/dialects/postgres/data-types.js]: Here it's 'parameter' the other places it's 'attribute', intentional?

**Comment by mickhansen** [comment]
Looks great. Warnings are the way to go, since i don't personally thing we should ever remove support completely.

File: lib/data-types.js (hunk 2 above)
Comment (janmeier) [inline lib/data-types.js]: Couldn't we actually just use depd here?

Comment (mickhansen) [inline lib/data-types.js]: It's not really a deprecation, i'm not sure if depd explicitely uses that wording though to.

Comment (BridgeAR) [inline lib/data-types.js]: Hm that would be nice indeed... It seems like depd [includes 'deprecated' hardcoded](https://github.com/dougwilson/nodejs-depd/blob/master/index.js#L100). I could include the colours without difficulties though :D

File: lib/dialects/postgres/data-types.js (hunk 3 above)
Comment (BridgeAR) [inline lib/dialects/postgres/data-types.js]: I'm unhappy with the wording but nothing better came to my mind. The difference I tried to emphasize is between `BLOB('tiny')` and `INTEGER.UNSIGNED`.

**Comment by BridgeAR** [comment]
Rebased, fixed a couple typos and included more warnings. Depending on what you wish I'd say it's ready to merge. We could provide the additional functionality to include the lines later on.

**Comment by mickhansen** [comment]
You're still mixing `attribute` and `parameter` in the warnings, would be nice to streamline them. Or provide some reasoning why they are different.

File: lib/dialects/postgres/data-types.js (hunk 3 above)
Comment (mickhansen) [inline lib/dialects/postgres/data-types.js]: I see, that difference isn't really obvious at all (to me atleast), so i'd say just use "parameter" for all.

File: lib/data-types.js (hunk 2 above)
Comment (mickhansen) [inline lib/data-types.js]: Could duplicate the stack functionality though.

Comment (BridgeAR) [inline lib/data-types.js]: I thought about it but let's move that to another day and another PR :)

File: lib/dialects/postgres/data-types.js (hunk 3 above)
Comment (BridgeAR) [inline lib/dialects/postgres/data-types.js]: Done

**Comment by mickhansen** [comment]
General wording be better with something like 'postgres does not support TEXT with size. Simple 'TEXT' will be used instead`

**Comment by BridgeAR** [comment]
+1

**Comment by BridgeAR** [comment]
Changed the comments and rebased

**Comment by diosney** [comment]
Please, see https://github.com/sequelize/sequelize/issues/6385

I really don't like this PR, there is a way to disable completely this messages?
